README.md

```diff
@@ -104,7 +124,73 @@ table names in the SQL query. The actual format after replacement will depend on
 used.
  
 
-# How to Add New Consumers
-Consumers should be added to the `tests/consumers` directory and provide 
+# Substrait Function Tests
+The substrait function tests aim to test the functions available in substrait.  This is done
+by converting queries (SQL/Ibis expressions) into substrait plans via various producers and
+running the substrait plans on different consumers.  The results of running the substrait plans
+are then compared to an expected result, which is determined by running the original query
+against an known engine (currently we use DuckDB).
+
+Substrait function test files are located in the `tests/functional/extension_functions` folder.
+
+
+## Test Case Args
+Test case arguments located in `tests/functional/queries/{*_tests}.py`.  They specify 
+the parquet files, a SQL query, and an ibis expression.
+
+The tests also take in the consumer and producer as test input via the producer/consumer test fixtures,
+which are defined in `test/conftest.py`.  The fixtures allow the tests to cycle through all combinations
+of producers and consumers.
+
+arithmetic_tests.py
+```python
+SCALAR_FUNCTIONS = (
+    {
+        "test_name": "add",
+        "file_names": ["partsupp.parquet"],
+        "sql_query": SQL_SCALAR["add"],
+        "ibis_expr": IBIS_SCALAR["add"],
+    },
+```
```

Comment:
What is this, an example?

Reply:
Yeah, this is supposed to be an example excerpt from that file.